lib/rules/function-url-quotes/index.mjs

```diff
@@ -158,24 +158,26 @@ const rule = (primary, secondaryOptions) => {
 		 * @returns {Pick<Problem, 'index' | 'endIndex'> & { messageType: keyof messages } | undefined}
 		 */
 		function checkArgs(args, index) {
-			const leftTrimmedArgs = args.trimStart();
+			const trimmedArgs = args.trim();
 
-			if (!isStandardSyntaxUrl(leftTrimmedArgs)) return;
+			if (!isStandardSyntaxUrl(trimmedArgs)) return;
 
 			let expectQuotes = primary === 'always';
 
-			if (exceptEmpty && emptyArgumentPatterns.has(args.trim())) {
+			if (exceptEmpty && emptyArgumentPatterns.has(trimmedArgs)) {
 				expectQuotes = !expectQuotes;
 			}
 
-			const hasQuotes = leftTrimmedArgs.startsWith("'") || leftTrimmedArgs.startsWith('"');
+			const hasQuotes = trimmedArgs.startsWith("'") || trimmedArgs.startsWith('"');
 
 			if (expectQuotes && hasQuotes) return;
 
 			if (!expectQuotes && !hasQuotes) return;
 
+			if (hasQuotes && trimmedArgs[0] !== trimmedArgs[trimmedArgs.length - 1]) return;
```

Comment:
Let's extract `trimmedArgs[0] !== trimmedArgs[trimmedArgs.length - 1])` into a function called `hasModifier(args)`, making the code self-documenting for future maintainers.

Reply:
I moved the check into `hasModifier(args)`, and in the next commit, I rewrote it. This was necessary for the fix